src/java.base/share/classes/java/lang/VirtualThread.java

```diff
@@ -991,7 +991,8 @@ private void setCarrierThread(Thread carrier) {
 
     // -- JVM TI support --
 
-    private static volatile boolean notifyJvmtiEvents;  // set by VM
+    // will always be true because a JVM TI agent can be attached at any time
+    private static volatile boolean notifyJvmtiEvents = true;
```

Comment:
Why do we want this change instead of having OpenJ9 do as the comment, "set by VM", says?

Reply:
When I tried to add the field to vmconstantpool I hit a build crash. https://github.com/eclipse-openj9/openj9/pull/15517#discussion_r918358904
I found that it was because during resolve the VirtualThread static initializer methods were trying to access system properties before they were initialized.
We discussed this in a Loom meeting and decided to set it to true in the jcl.

Reply:
There is another to set the field in the VM:
```
    jclass vthreadClass = (*env)->FindClass(env, "java/lang/VirtualThread");
    if (NULL != vthreadClass) {
        jfieldID vthreadFieldID = (*env)->GetStaticFieldID(env, vthreadClass, "notifyJvmtiEvents", "Z");
        if (NULL != vthreadFieldID) {
            (*env)->SetStaticBooleanField(env, vthreadClass, vthreadField, JNI_TRUE);
        }
    }
```

Reply:
`java -verbose:class -version` shows `VirtualThread` being loaded; I expect it will always be loaded anyway.
If `VirtualThread` can be added to `vmconstantpool.xml`, then the class lookup would be less expensive.

Reply:
https://github.com/eclipse-openj9/openj9/pull/15690 will introduce `VirtualThread` to `vmconstantpool.xml`. The JNI approach will also work fine for us.

Reply:
I wasn't suggesting that your proposal wouldn't work, I was just pointing out a way to reduce of the cost of that initialization.

Reply:
Actually this will need to be done before a virtual thread can be run since its natives need to be enabled at virtual thread start

Reply:
See https://github.com/eclipse-openj9/openj9/blob/4235965a052358dfd0c645a036db86674b487efe/runtime/jcl/common/jclcinit.c#L739-L745 (`String.compressionFlag`).

Reply:
> String.compressionFlag

That is added to the vmconstantpool which we can't do otherwise we get the crash

Reply:
I believe that the `initializeStaticField` call is needed along with adding the static field to the `vmconstantpool` for resolving the crash. In `jclinit.c`, this approach is used for all other static fields, which are introduced in `vmconstantpool`.

Reply:
I am still getting the crash with initializeStaticField.